tools/gce-usage-log/README.md

```diff
@@ -0,0 +1,227 @@
+
+The resulting schema of the view looks like this:
+
+![view schema](images/view-schema.png)
+
+### 1.4 Component Architecture
+
+Bringing it all together, we can visualize how the components interact with this diagram.
+
+![architecture diagram](images/gce-usage-log.png)
+
+## 2. Setup
+
+Let’s go through the steps to configure audit logging, scan your VM inventory, and create the BigQuery view.
+
+We recommend operating in the Cloud Shell as it already has the necessary tools installed.
+
+### 2.1 Prerequisites
+
+Before starting, let’s gather some prerequisite information.
+
+* PROJECT_ID: The ID of the project
+* BQ_LOCATON: BigQuery location to use for your dataset
+* ORG_NUMBER: The numeric id of the organization. Find yours by invoking `gcloud organizations list`
+
+Now export them as environment variables in your Cloud Shell for subsequent use.
+
+```bash
+export PROJECT_ID=myusageproject
+export BQ_LOCATION=EU
+export ORG_NUMBER=622243302570
+```
+
+### 2.2 Permissions
+
+To follow these interactive instructions, you will need permissions to:
+
+* Create and manage a BQ dataset in the project you’ve chosen
+  * bigquery.datasets.create
+  * bigquery.tables.create
+  * bigquery.tables.getData
+  * bigquery.tables.updateData
+
+And you will need permissions at the organization level to:
+
+* Create an organization-level audit log sink
+  * roles/logging.configWriter
+* Scan your inventory:
+  * compute.disks.get
+  * compute.instances.get
+  * compute.instances.list
+  * compute.zones.get
+  * compute.zones.list
+  * resourcemanager.organizations.get
+  * resourcemanager.projects.get
+  * resourcemanager.projects.list
+
+You will need these permissions only during initial setup.
+
+### 2.3 Preparing the audit log dataset
+
+Let’s walk through the steps necessary to create and populate the dataset.
+
+#### 2.3.1 Creating the dataset
+
+Now, let’s create the dataset inside the project.
+
+```bash
+bq mk --location=${BQ_LOCATION} -d "${PROJECT_ID}:gce_usage_log"
+```
+
+#### 2.3.2 Create the audit log sink
+
+```bash
+gcloud logging sinks create gce_usage \
+  bigquery.googleapis.com/projects/${PROJECT_ID}/datasets/gce_usage_log \
+  --log-filter='resource.type="gce_instance" AND
+  protoPayload.serviceName="compute.googleapis.com"' \
+  --organization=${ORG_NUMBER} --include-children
+```
+
+This command will create and return a service account ID, such as serviceAccount:o125240632470-886280@gcp-sa-logging.iam.gserviceaccount.com.
+
+Note down the account name, as we'll use it in the next step!
+
+#### 2.3.3 Allow the service account to write to our dataset
+
+Taking the service account created in the last step, let’s assign to it the narrowest privileges on our dataset - in this case, “BigQuery Data Editor.”
+
+This can be done a number of ways, and here we’ll just follow [the manual process](https://cloud.google.com/bigquery/docs/dataset-access-controls#bigquery_update_dataset_access-console).
```

Comment:
Probably replace this with the full link to the instructions https://cloud.google.com/bigquery/docs/dataset-access-controls#controlling_access_to_a_dataset

Reply:
Good suggestion - fixed.